plugin/src/main/scala/chisel3/internal/plugin/BundleComponent.scala

```diff
@@ -63,62 +97,177 @@ private[plugin] class BundleComponent(val global: Global, arguments: ChiselPlugi
+
+            def isBundleField(member: Symbol): Boolean = {
+              if (!member.isAccessor) {
+//                indentShow(s"- $member is not accessor")
+                false
+              } else if (isData(member.tpe.typeSymbol)) {
+                indentShow(s"+ $member is Data")
+                true
+              } else if (isOptionOfData(member)) {
+                indentShow(s"+ $member is Option[Data]")
+                true
+              } else {
+//                indentShow(s"- $member not Data or Option[Data]")
+                false
+              }
+            }
+
+            indentShow(s"getAll: Current Bundle: ${bundleSymbol.name}")
+
+            val currentFields = bundleSymbol.info.members.flatMap {
+
+              case member if member.isPublic =>
+                if (isBundleField(member)) {
+                  // The params have spaces after them (Scalac implementation detail)
+                  Some(member.name.toString.trim -> gen.mkAttributedSelect(thiz.asInstanceOf[Tree], member))
+                } else {
+                  //TODO: The following code block creates a compiler error for bundle fields that are
+                  //      Seq[Data]. The test for this case is a bit too hacky so for now we will just
+                  //      elide these fields.
+//                  if (member.isAccessor && typeIsSeqOfData(member.tpe) && !isIgnoreSeqInBundle(bundleSymbol)) {
+//                    global.reporter.error(
+//                      member.pos,
+//                      s"Bundle.field ${bundleSymbol.name}.${member.name} cannot be a Seq[Data]. " +
+//                        "Use Vec or MixedVec or mix in trait IgnoreSeqInBundle"
+//                    )
+//                  }
+                  None
+                }
+
+              case _ => None
+            }.toList
+
+            indentShow(s"\nstarting parents: ${bundleSymbol.parentSymbols.mkString(",")}")
+            val allParentFields = bundleSymbol.parentSymbols.flatMap { parentSymbol =>
+              val fieldsFromParent = if (depth < 1 && !isExactBundle(bundleSymbol)) {
+                indentShow(s"Calling getAll($parentSymbol) ")
+                val foundFields = getAllBundleFields(parentSymbol, depth + 1)
+                indentShow(s"Return getAll($parentSymbol): " + foundFields.map(_._1).mkString(","))
+                foundFields
+              } else {
+                List()
+              }
+              fieldsFromParent
+            }
+            allParentFields ++ currentFields
+          }
+
+          val elementArgs = getAllBundleFields(bundle.symbol)
+
+          val elementsImplSym =
+            bundle.symbol.newMethod(TermName("_elementsImpl"), bundle.symbol.pos.focus, Flag.OVERRIDE | Flag.PROTECTED)
+          // Handwritten cloneTypes don't have the Method flag set, unclear if it matters
+          elementsImplSym.resetFlag(Flags.METHOD)
+          // Need to set the type to chisel3.Bundle for the override to work
```

Comment:
I believe these comments are wrong

Reply:
These comments were part of the code previous to this PR.
I just moved them. What do you suggest here?

Reply:
It looks more like you copy-pasted them because that comment is correct for `_cloneTypeImpl` but not correct for `_elementsImpl`, the return type is not `chisel3.Bundle`, it's `SeqMap[String, Any]` (see next line).

Reply:
remove comments